### ISSUE TITLE
Bad address index.

### ISSUE THREAD

**Issue by rr13k** [godotengine/godot#80714]
### Godot version

4.1

### System information

macOS12.1

### Issue description

func test():
	if self is Player:
		pass

Impossible code that prevents it from running

### Steps to reproduce

run main.tscn

### Minimal reproduction project

[bug.zip](https://github.com/godotengine/godot/files/12368994/bug.zip)

**Comment by rr13k** [comment]
supplement  The following modifications still have the same error

if is_instance_of(self,Player):

**Comment by rr13k** [comment]
error image:
![image](https://github.com/godotengine/godot/assets/45162461/e6b1a435-6157-4d4f-889b-cf579233fdf2)

**Comment by bitsawer** [comment]
I can repro this in Godot 4.1.1, but looks like this has already been fixed in current 4.2 master (0511f9d9a7d56c742d87fafdcea8785d40ad14b3) and 4.2 dev 3 release. Closing, but feel free to comment or reopen the issue if the problem is not solved for some reason.

**Comment by dalexeev** [comment]
I reproduced in 4.1.1:

![](https://github.com/godotengine/godot/assets/47700418/b75e814f-683c-48f6-ac4f-fd064e8eb2a6)

**Comment by dalexeev** [comment]
> Possible duplicate of https://github.com/godotengine/godot/issues/74253?

Yes, the reason is the same, circular dependencies/requirements:

```mermaid
flowchart LR
    main.gd-- requires -->char.tscn
    char.tscn-- requires -->char.gd
    char.gd-- extends -->entry.gd
    entry.gd-- requires -->player.gd
    player.gd-- extends -->char.gd
```

> I did mention that I can also reproduce this in 4.1.1

Sorry, I misread.

> but this has been fixed at some point since (fixed at least in 4.2 dev 3 relese) so I don't think this is a valid issue anymore

> Although it could be useful to find out what PR fixed this for future reference

I don't think this has been properly fixed, it's a complex issue. However, we can close this as a duplicate of #74253.
